[BUG] xdeepfm error in AzureML test

### Description
<!--- Describe your issue/bug/request in detail -->
```
    @pytest.mark.gpu
    @pytest.mark.notebooks
    @pytest.mark.integration
    @pytest.mark.parametrize(
        "syn_epochs, criteo_epochs, expected_values, seed",
        [
            (
                15,
                10,
                ***
                    "res_syn": ***"auc": 0.9716, "logloss": 0.699***,
                    "res_real": ***"auc": 0.749, "logloss": 0.4926***,
                ***,
                42,
            )
        ],
    )
    def test_xdeepfm_integration(
        notebooks,
        output_notebook,
        kernel_name,
        syn_epochs,
        criteo_epochs,
        expected_values,
        seed,
    ):
        notebook_path = notebooks["xdeepfm_quickstart"]
        pm.execute_notebook(
            notebook_path,
            output_notebook,
            kernel_name=kernel_name,
            parameters=dict(
                EPOCHS_FOR_SYNTHETIC_RUN=syn_epochs,
                EPOCHS_FOR_CRITEO_RUN=criteo_epochs,
                BATCH_SIZE_SYNTHETIC=1024,
                BATCH_SIZE_CRITEO=1024,
                RANDOM_SEED=seed,
            ),
        )
        results = sb.read_notebook(output_notebook).scraps.dataframe.set_index("name")[
            "data"
        ]
    
        for key, value in expected_values.items():
>           assert results[key]["auc"] == pytest.approx(value["auc"], rel=TOL, abs=ABS_TOL)
E           assert 0.5131 == 0.9716 ± 9.7e-02
E             comparison failed
E             Obtained: 0.5131
E             Expected: 0.9716 ± 9.7e-02
```

### In which platform does it happen?
<!--- Describe the platform where the issue is happening (use a list if needed) -->
<!--- For example: -->
<!--- * Azure Data Science Virtual Machine. -->
<!--- * Azure Databricks.  -->
<!--- * Other platforms.  -->

### How do we replicate the issue?
<!--- Please be specific as possible (use a list if needed). -->
<!--- For example: -->
<!--- * Create a conda environment for pyspark -->
<!--- * Run unit test `test_sar_pyspark.py` with `pytest -m 'spark'` -->
<!--- * ... -->
See https://github.com/microsoft/recommenders/actions/runs/3459763061/jobs/5775521889


### Expected behavior (i.e. solution)
<!--- For example:  -->
<!--- * The tests for SAR PySpark should pass successfully. -->

### Other Comments

--- a/examples/00_quick_start/xdeepfm_criteo.ipynb
+++ b/examples/00_quick_start/xdeepfm_criteo.ipynb
@@ -331,7 +331,8 @@
     }
    ],
    "source": [
-    "sb.glue(\"result\", result)"
+    "sb.glue(\"auc\", result[\"auc\"])\n",
+    "sb.glue(\"logloss\", result[\"logloss\"])"
    ]
   },
   {

--- a/examples/00_quick_start/xdeepfm_criteo.ipynb
+++ b/examples/00_quick_start/xdeepfm_criteo.ipynb
@@ -20,7 +20,7 @@
     "* It contains a traditional DNN component that learns feature interactions in an implicit fashion and in bit-wise level.\n",
     "* The implementation makes this model quite configurable. We can enable different subsets of components by setting hyperparameters like `use_Linear_part`, `use_FM_part`, `use_CIN_part`, and `use_DNN_part`. For example, by enabling only the `use_Linear_part` and `use_FM_part`, we can get a classical FM model.\n",
     "\n",
-    "In this notebook, we test xDeepFM on two datasets: 1) a small synthetic dataset and 2) [Criteo dataset](http://labs.criteo.com/category/dataset)"
+    "In this notebook, we test xDeepFM on [Criteo dataset](http://labs.criteo.com/category/dataset)."
    ]
   },
   {
@@ -36,20 +36,12 @@
    "metadata": {},
    "outputs": [
     {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "/anaconda/envs/tf2/lib/python3.7/site-packages/papermill/iorw.py:50: FutureWarning: pyarrow.HadoopFileSystem is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.\n",
-      "  from pyarrow import HadoopFileSystem\n"
-     ]
-    },
-    {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "System version: 3.7.11 (default, Jul 27 2021, 14:32:16) \n",
-      "[GCC 7.5.0]\n",
-      "Tensorflow version: 2.6.1\n"
+      "System version: 3.7.13 (default, Oct 18 2022, 18:57:03) \n",
+      "[GCC 11.2.0]\n",
+      "Tensorflow version: 2.7.4\n"
      ]
     }
    ],
@@ -61,7 +53,6 @@
     "import tensorflow as tf\n",
     "tf.get_logger().setLevel('ERROR') # only show error messages\n",
     "\n",
-    "from recommenders.utils.constants import SEED\n",
     "from recommenders.models.deeprec.deeprec_utils import (\n",
     "    download_deeprec_resources, prepare_hparams\n",
     ")\n",
@@ -89,11 +80,9 @@
    },
    "outputs": [],
    "source": [
-    "EPOCHS_FOR_SYNTHETIC_RUN = 15\n",
-    "EPOCHS_FOR_CRITEO_RUN = 10\n",
-    "BATCH_SIZE_SYNTHETIC = 128\n",
-    "BATCH_SIZE_CRITEO = 4096\n",
-    "RANDOM_SEED = SEED  # Set to None for non-deterministic result\n"
+    "EPOCHS = 10\n",
+    "BATCH_SIZE = 1024\n",
+    "RANDOM_SEED = 42  # Set this to None for non-deterministic result\n"
    ]
   },
   {
@@ -102,10 +91,7 @@
    "source": [
     "xDeepFM uses the FFM format as data input: `<label> <field_id>:<feature_id>:<feature_value>`  \n",
     "Each line represents an instance, `<label>` is a binary value with 1 meaning positive instance and 0 meaning negative instance. \n",
-    "Features are divided into fields. For example, user's gender is a field, it contains three possible values, i.e. male, female and unknown. Occupation can be another field, which contains many more possible values than the gender field. Both field index and feature index are starting from 1. <br>\n",
-    "\n",
-    "## 1. Synthetic data\n",
-    "Now let's start with a small synthetic dataset. In this dataset, there are 10 fields, 1000 fefatures, and label is generated according to the result of a set of preset pair-wise feature interactions. "
+    "Features are divided into fields. For example, user's gender is a field, it contains three possible values, i.e. male, female and unknown. Occupation can be another field, which contains many more possible values than the gender field. Both field index and feature index are starting from 1. <br>"
    ]
   },
   {
@@ -117,7 +103,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "100%|██████████| 10.3k/10.3k [00:01<00:00, 5.24kKB/s]\n"
+      "100%|██████████| 10.3k/10.3k [00:01<00:00, 6.89kKB/s]\n"
      ]
     }
    ],
@@ -125,10 +111,10 @@
     "tmpdir = TemporaryDirectory()\n",
     "data_path = tmpdir.name\n",
     "yaml_file = os.path.join(data_path, r'xDeepFM.yaml')\n",
-    "train_file = os.path.join(data_path, r'synthetic_part_0')\n",
-    "valid_file = os.path.join(data_path, r'synthetic_part_1')\n",
-    "test_file = os.path.join(data_path, r'synthetic_part_2')\n",
     "output_file = os.path.join(data_path, r'output.txt')\n",
+    "train_file = os.path.join(data_path, r'cretio_tiny_train')\n",
+    "valid_file = os.path.join(data_path, r'cretio_tiny_valid')\n",
+    "test_file = os.path.join(data_path, r'cretio_tiny_test')\n",
     "\n",
     "if not os.path.exists(yaml_file):\n",
     "    download_deeprec_resources(r'https://recodatasets.z20.web.core.windows.net/deeprec/', data_path, 'xdeepfmresources.zip')\n"
@@ -138,312 +124,28 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "#### 1.1 Prepare hyper-parameters\n",
-    "prepare_hparams() will create a full set of hyper-parameters for model training, such as learning rate, feature number, and dropout ratio. We can put those parameters in a yaml file, or pass parameters as the function's parameters (which will overwrite yaml settings)."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 4,
-   "metadata": {
-    "scrolled": true
-   },
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "kg_file=None,user_clicks=None,FEATURE_COUNT=1000,FIELD_COUNT=10,data_format=ffm,PAIR_NUM=None,DNN_FIELD_NUM=None,n_user=None,n_item=None,n_user_attr=None,n_item_attr=None,iterator_type=None,SUMMARIES_DIR=None,MODEL_DIR=None,wordEmb_file=None,entityEmb_file=None,contextEmb_file=None,news_feature_file=None,user_history_file=None,use_entity=True,use_context=True,doc_size=None,history_size=None,word_size=None,entity_size=None,entity_dim=None,entity_embedding_method=None,transform=None,train_ratio=None,dim=10,layer_sizes=[100, 100],cross_layer_sizes=[1],cross_layers=None,activation=['relu', 'relu'],cross_activation=identity,user_dropout=False,dropout=[0.0, 0.0],attention_layer_sizes=None,attention_activation=None,attention_dropout=0.0,model_type=xDeepFM,method=classification,load_saved_model=False,load_model_name=you model path,filter_sizes=None,num_filters=None,mu=None,fast_CIN_d=0,use_Linear_part=False,use_FM_part=False,use_CIN_part=True,use_DNN_part=False,init_method=tnormal,init_value=0.3,embed_l2=0.0001,embed_l1=0.0,layer_l2=0.0001,layer_l1=0.0,cross_l2=0.0001,cross_l1=0.0,reg_kg=0.0,learning_rate=0.001,lr_rs=1,lr_kg=0.5,kg_training_interval=5,max_grad_norm=2,is_clip_norm=0,dtype=32,loss=log_loss,optimizer=adam,epochs=15,batch_size=128,enable_BN=False,show_step=200000,save_model=False,save_epoch=2,metrics=['auc', 'logloss'],write_tfevents=False,item_embedding_dim=None,cate_embedding_dim=None,user_embedding_dim=None,train_num_ngs=4,need_sample=True,embedding_dropout=0.0,user_vocab=None,item_vocab=None,cate_vocab=None,pairwise_metrics=None,EARLY_STOP=100,max_seq_length=None,hidden_size=None,L=None,T=None,n_v=None,n_h=None,min_seq_length=1,attention_size=None,att_fcn_layer_sizes=None,dilations=None,kernel_size=None,embed_size=None,n_layers=None,decay=None,eval_epoch=None,top_k=None\n"
-     ]
-    }
-   ],
-   "source": [
-    "hparams = prepare_hparams(yaml_file, \n",
-    "                          FEATURE_COUNT=1000, \n",
-    "                          FIELD_COUNT=10, \n",
-    "                          cross_l2=0.0001, \n",
-    "                          embed_l2=0.0001, \n",
-    "                          learning_rate=0.001, \n",
-    "                          epochs=EPOCHS_FOR_SYNTHETIC_RUN,\n",
-    "                          batch_size=BATCH_SIZE_SYNTHETIC)\n",
-    "print(hparams)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "#### 1.2 Create data loader\n",
-    "Designate a data iterator for the model. xDeepFM uses FFMTextIterator. "
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 5,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "input_creator = FFMTextIterator"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "#### 1.3 Create model\n",
-    "When both hyper-parameters and data iterator are ready, we can create a model:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 6,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "Add CIN part.\n"
-     ]
-    }
-   ],
-   "source": [
-    "model = XDeepFMModel(hparams, input_creator, seed=RANDOM_SEED)\n",
-    "\n",
-    "## sometimes we don't want to train a model from scratch\n",
-    "## then we can load a pre-trained model like this: \n",
-    "#model.load_model(r'your_model_path')"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Now let's see what is the model's performance at this point (without starting training):"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 7,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "{'auc': 0.5043, 'logloss': 0.7515}\n"
-     ]
-    }
-   ],
-   "source": [
-    "print(model.run_eval(test_file))"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "AUC=0.5 is a state of random guess. We can see that before training, the model behaves like random guessing.\n",
-    "\n",
-    "#### 1.4 Train model\n",
-    "Next we want to train the model on a training set, and check the performance on a validation dataset. Training the model is as simple as a function call:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 8,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "at epoch 1\n",
-      "train info: logloss loss:0.7556826068773302\n",
-      "eval info: auc:0.504, logloss:0.7042\n",
-      "at epoch 1 , train time: 4.3 eval time: 0.6\n",
-      "at epoch 2\n",
-      "train info: logloss loss:0.7263523231666932\n",
-      "eval info: auc:0.5066, logloss:0.6973\n",
-      "at epoch 2 , train time: 4.3 eval time: 0.8\n",
-      "at epoch 3\n",
-      "train info: logloss loss:0.7177084291104189\n",
-      "eval info: auc:0.5099, logloss:0.6953\n",
-      "at epoch 3 , train time: 3.8 eval time: 0.7\n",
-      "at epoch 4\n",
-      "train info: logloss loss:0.7118660186983875\n",
-      "eval info: auc:0.5147, logloss:0.6946\n",
-      "at epoch 4 , train time: 3.7 eval time: 0.8\n",
-      "at epoch 5\n",
-      "train info: logloss loss:0.7055103289302682\n",
-      "eval info: auc:0.523, logloss:0.6941\n",
-      "at epoch 5 , train time: 3.8 eval time: 0.7\n",
-      "at epoch 6\n",
-      "train info: logloss loss:0.6954095556154284\n",
-      "eval info: auc:0.5416, logloss:0.6929\n",
-      "at epoch 6 , train time: 3.6 eval time: 0.8\n",
-      "at epoch 7\n",
-      "train info: logloss loss:0.6723950118133702\n",
-      "eval info: auc:0.5916, logloss:0.6831\n",
-      "at epoch 7 , train time: 4.0 eval time: 0.7\n",
-      "at epoch 8\n",
-      "train info: logloss loss:0.6119807973964927\n",
-      "eval info: auc:0.7024, logloss:0.6288\n",
-      "at epoch 8 , train time: 4.2 eval time: 0.8\n",
-      "at epoch 9\n",
-      "train info: logloss loss:0.5020270644594303\n",
-      "eval info: auc:0.8154, logloss:0.5257\n",
-      "at epoch 9 , train time: 4.3 eval time: 0.8\n",
-      "at epoch 10\n",
-      "train info: logloss loss:0.38994721433346213\n",
-      "eval info: auc:0.8826, logloss:0.4315\n",
-      "at epoch 10 , train time: 4.4 eval time: 0.7\n",
-      "at epoch 11\n",
-      "train info: logloss loss:0.30144522643785704\n",
-      "eval info: auc:0.9205, logloss:0.3605\n",
-      "at epoch 11 , train time: 4.4 eval time: 0.6\n",
-      "at epoch 12\n",
-      "train info: logloss loss:0.23429049589892023\n",
-      "eval info: auc:0.9431, logloss:0.3082\n",
-      "at epoch 12 , train time: 4.2 eval time: 0.6\n",
-      "at epoch 13\n",
-      "train info: logloss loss:0.18283964168677216\n",
-      "eval info: auc:0.9577, logloss:0.2682\n",
-      "at epoch 13 , train time: 3.9 eval time: 0.7\n",
-      "at epoch 14\n",
-      "train info: logloss loss:0.14280925843467826\n",
-      "eval info: auc:0.9676, logloss:0.2369\n",
-      "at epoch 14 , train time: 3.9 eval time: 0.7\n",
-      "at epoch 15\n",
-      "train info: logloss loss:0.11175788457655825\n",
-      "eval info: auc:0.9745, logloss:0.2128\n",
-      "at epoch 15 , train time: 3.7 eval time: 0.7\n"
-     ]
-    },
-    {
-     "data": {
-      "text/plain": [
-       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7f9d74f7ff60>"
-      ]
-     },
-     "execution_count": 8,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
-   "source": [
-    "model.fit(train_file, valid_file)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "#### 1.5 Evaluate model\n",
-    "\n",
-    "Again, let's see what is the model's performance now (after training):"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 9,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "{'auc': 0.9716, 'logloss': 0.2278}\n"
-     ]
-    }
-   ],
-   "source": [
-    "res_syn = model.run_eval(test_file)\n",
-    "print(res_syn)\n"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 10,
-   "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "application/scrapbook.scrap.json+json": {
-       "data": {
-        "auc": 0.9716,
-        "logloss": 0.2278
-       },
-       "encoder": "json",
-       "name": "res_syn",
-       "version": 1
-      }
-     },
-     "metadata": {
-      "scrapbook": {
-       "data": true,
-       "display": false,
-       "name": "res_syn"
-      }
-     },
-     "output_type": "display_data"
-    }
-   ],
-   "source": [
-    "sb.glue(\"res_syn\", res_syn)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "If we want to get the full prediction scores rather than evaluation metrics, we can do this:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 11,
-   "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/plain": [
-       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7f9d74f7ff60>"
-      ]
-     },
-     "execution_count": 11,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
-   "source": [
-    "model.predict(test_file, output_file)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
     "## 2. Criteo data \n",
     "\n",
-    "Now we have successfully launched an experiment on a synthetic dataset. Next let's try something on a real world dataset, which is a small sample from [Criteo dataset](http://labs.criteo.com/category/dataset). Criteo dataset is a well known industry benchmarking dataset for developing CTR prediction models and it's frequently adopted as evaluation dataset by research papers. The original dataset is too large for a lightweight demo, so we sample a small portion from it as a demo dataset."
+    "Now let's try the xDeepFM on a real world dataset, a small sample from [Criteo dataset](http://labs.criteo.com/category/dataset). Criteo dataset is a well known industry benchmarking dataset for developing CTR prediction models and it's frequently adopted as evaluation dataset by research papers. \n",
+    "\n",
+    "The original dataset is too large for a lightweight demo, so we sample a small portion from it as a demo dataset."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 12,
+   "execution_count": 13,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "demo with Criteo dataset\n"
+      "Demo with Criteo dataset\n"
      ]
     }
    ],
    "source": [
-    "print('demo with Criteo dataset')\n",
+    "print('Demo with Criteo dataset')\n",
     "hparams = prepare_hparams(yaml_file, \n",
     "                          FEATURE_COUNT=2300000, \n",
     "                          FIELD_COUNT=39, \n",
@@ -464,18 +166,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 13,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "train_file = os.path.join(data_path, r'cretio_tiny_train')\n",
-    "valid_file = os.path.join(data_path, r'cretio_tiny_valid')\n",
-    "test_file = os.path.join(data_path, r'cretio_tiny_test')"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 14,
+   "execution_count": 16,
    "metadata": {},
    "outputs": [
     {
@@ -486,6 +177,13 @@
       "Add CIN part.\n",
       "Add DNN part.\n"
      ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-11-16 11:12:42.279935: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 15397 MB memory:  -> device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0001:00:00.0, compute capability: 6.0\n"
+     ]
     }
    ],
    "source": [
@@ -494,7 +192,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 15,
+   "execution_count": 17,
    "metadata": {},
    "outputs": [
     {
@@ -507,12 +205,12 @@
    ],
    "source": [
     "# check the predictive performance before the model is trained\n",
-    "print(model.run_eval(test_file)) \n"
+    "print(model.run_eval(test_file)) "
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 16,
+   "execution_count": 18,
    "metadata": {},
    "outputs": [
     {
@@ -520,65 +218,66 @@
      "output_type": "stream",
      "text": [
       "at epoch 1\n",
-      "train info: logloss loss:744.3602027893066\n",
+      "train info: logloss loss:744.3602104187012\n",
       "eval info: auc:0.6637, logloss:0.5342\n",
-      "at epoch 1 , train time: 21.7 eval time: 4.3\n",
+      "at epoch 1 , train time: 20.0 eval time: 3.9\n",
       "at epoch 2\n",
-      "train info: logloss loss:385.66927337646484\n",
+      "train info: logloss loss:385.66929054260254\n",
       "eval info: auc:0.7137, logloss:0.5109\n",
-      "at epoch 2 , train time: 21.4 eval time: 4.3\n",
+      "at epoch 2 , train time: 19.2 eval time: 4.0\n",
       "at epoch 3\n",
-      "train info: logloss loss:191.50830841064453\n",
+      "train info: logloss loss:191.5083179473877\n",
       "eval info: auc:0.7283, logloss:0.5037\n",
-      "at epoch 3 , train time: 21.4 eval time: 4.2\n",
+      "at epoch 3 , train time: 19.5 eval time: 3.9\n",
       "at epoch 4\n",
-      "train info: logloss loss:92.20774269104004\n",
+      "train info: logloss loss:92.20774817466736\n",
       "eval info: auc:0.7359, logloss:0.4991\n",
-      "at epoch 4 , train time: 21.6 eval time: 4.4\n",
+      "at epoch 4 , train time: 19.4 eval time: 3.9\n",
       "at epoch 5\n",
-      "train info: logloss loss:43.159456968307495\n",
+      "train info: logloss loss:43.15945792198181\n",
       "eval info: auc:0.74, logloss:0.4963\n",
-      "at epoch 5 , train time: 21.6 eval time: 4.3\n",
+      "at epoch 5 , train time: 19.6 eval time: 3.9\n",
       "at epoch 6\n",
-      "train info: logloss loss:19.656921446323395\n",
+      "train info: logloss loss:19.656923294067383\n",
       "eval info: auc:0.7426, logloss:0.4946\n",
-      "at epoch 6 , train time: 21.3 eval time: 4.2\n",
+      "at epoch 6 , train time: 19.7 eval time: 3.9\n",
       "at epoch 7\n",
-      "train info: logloss loss:8.77035716176033\n",
+      "train info: logloss loss:8.770357578992844\n",
       "eval info: auc:0.7441, logloss:0.4934\n",
-      "at epoch 7 , train time: 21.5 eval time: 4.3\n",
+      "at epoch 7 , train time: 19.6 eval time: 4.0\n",
       "at epoch 8\n",
-      "train info: logloss loss:3.9227354377508163\n",
+      "train info: logloss loss:3.9227356910705566\n",
       "eval info: auc:0.7453, logloss:0.4925\n",
-      "at epoch 8 , train time: 21.7 eval time: 4.3\n",
+      "at epoch 8 , train time: 19.8 eval time: 3.9\n",
       "at epoch 9\n",
-      "train info: logloss loss:1.8598770573735237\n",
+      "train info: logloss loss:1.8598771467804909\n",
       "eval info: auc:0.7462, logloss:0.4917\n",
-      "at epoch 9 , train time: 21.3 eval time: 4.2\n",
+      "at epoch 9 , train time: 19.5 eval time: 4.0\n",
       "at epoch 10\n",
-      "train info: logloss loss:1.0249397158622742\n",
+      "train info: logloss loss:1.0249397829174995\n",
       "eval info: auc:0.747, logloss:0.491\n",
-      "at epoch 10 , train time: 21.5 eval time: 4.3\n"
+      "at epoch 10 , train time: 19.7 eval time: 3.9\n"
      ]
     },
     {
      "data": {
       "text/plain": [
-       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7f9d64b4a2e8>"
+       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7fa2146f26d0>"
       ]
      },
-     "execution_count": 16,
+     "execution_count": 18,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
+    "%%time\n",
     "model.fit(train_file, valid_file)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 17,
+   "execution_count": 19,
    "metadata": {},
    "outputs": [
     {
@@ -591,8 +290,8 @@
    ],
    "source": [
     "# check the predictive performance after the model is trained\n",
-    "res_real = model.run_eval(test_file)\n",
-    "print(res_real)"
+    "result = model.run_eval(test_file)\n",
+    "print(result)"
    ]
   },
   {
@@ -623,7 +322,7 @@
     }
    ],
    "source": [
-    "sb.glue(\"res_real\", res_real)"
+    "sb.glue(\"result\", result)"
    ]
   },
   {
@@ -651,8 +350,9 @@
    "hash": "3a9a0c422ff9f08d62211b9648017c63b0a26d2c935edc37ebb8453675d13bb5"
   },
   "kernelspec": {
-   "display_name": "Python 3.7.11 64-bit ('tf2': conda)",
-   "name": "python3"
+   "display_name": "reco_gpu",
+   "language": "python",
+   "name": "conda-env-reco_gpu-py"
   },
   "language_info": {
    "codemirror_mode": {
@@ -664,9 +364,9 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.11"
+   "version": "3.7.13"
   }
  },
  "nbformat": 4,
- "nbformat_minor": 2
+ "nbformat_minor": 4
 }

--- a/examples/00_quick_start/xdeepfm_criteo.ipynb
+++ b/examples/00_quick_start/xdeepfm_criteo.ipynb
@@ -79,7 +79,7 @@
    "outputs": [],
    "source": [
     "EPOCHS = 10\n",
-    "BATCH_SIZE = 1024\n",
+    "BATCH_SIZE = 4096\n",
     "RANDOM_SEED = 42  # Set this to None for non-deterministic result\n"
    ]
   },
@@ -101,7 +101,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "100%|██████████| 10.3k/10.3k [00:01<00:00, 7.08kKB/s]\n"
+      "100%|██████████| 10.3k/10.3k [00:01<00:00, 7.96kKB/s]\n"
      ]
     }
    ],
@@ -131,14 +131,15 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 4,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "Demo with Criteo dataset\n"
+      "Demo with Criteo dataset\n",
+      "HParams object with values {'use_entity': True, 'use_context': True, 'cross_activation': 'identity', 'user_dropout': False, 'dropout': [0.0, 0.0], 'attention_dropout': 0.0, 'load_saved_model': False, 'fast_CIN_d': 0, 'use_Linear_part': True, 'use_FM_part': False, 'use_CIN_part': True, 'use_DNN_part': True, 'init_method': 'tnormal', 'init_value': 0.1, 'embed_l2': 0.01, 'embed_l1': 0.0, 'layer_l2': 0.01, 'layer_l1': 0.0, 'cross_l2': 0.01, 'cross_l1': 0.0, 'reg_kg': 0.0, 'learning_rate': 0.002, 'lr_rs': 1, 'lr_kg': 0.5, 'kg_training_interval': 5, 'max_grad_norm': 2, 'is_clip_norm': 0, 'dtype': 32, 'optimizer': 'adam', 'epochs': 10, 'batch_size': 4096, 'enable_BN': False, 'show_step': 200000, 'save_model': False, 'save_epoch': 2, 'write_tfevents': False, 'train_num_ngs': 4, 'need_sample': True, 'embedding_dropout': 0.0, 'EARLY_STOP': 100, 'min_seq_length': 1, 'slots': 5, 'cell': 'SUM', 'FIELD_COUNT': 39, 'FEATURE_COUNT': 2300000, 'data_format': 'ffm', 'load_model_name': 'you model path', 'method': 'classification', 'model_type': 'xDeepFM', 'dim': 10, 'layer_sizes': [20, 20], 'activation': ['relu', 'relu'], 'cross_layer_sizes': [20, 10], 'loss': 'log_loss', 'metrics': ['auc', 'logloss']}\n"
      ]
     }
    ],
@@ -158,7 +159,8 @@
     "                          layer_sizes=[20,20],\n",
     "                          use_Linear_part=True, \n",
     "                          use_CIN_part=True, \n",
-    "                          use_DNN_part=True)\n"
+    "                          use_DNN_part=True)\n",
+    "print(hparams)"
    ]
   },
   {
@@ -179,11 +181,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "/anaconda/envs/reco_gpu/lib/python3.7/site-packages/tensorflow/python/framework/indexed_slices.py:450: UserWarning: Converting sparse IndexedSlices(IndexedSlices(indices=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Reshape_1:0\", shape=(None,), dtype=int32), values=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Reshape:0\", shape=(None, 10), dtype=float32), dense_shape=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Cast:0\", shape=(2,), dtype=int32))) to a dense Tensor of unknown shape. This may consume a large amount of memory.\n",
-      "  \"shape. This may consume a large amount of memory.\" % value)\n",
-      "2022-11-16 11:23:52.751265: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
-      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
-      "2022-11-16 11:23:54.342328: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 15397 MB memory:  -> device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0001:00:00.0, compute capability: 6.0\n"
+      "2022-11-16 11:30:58.632305: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 15397 MB memory:  -> device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0001:00:00.0, compute capability: 6.0\n"
      ]
     }
    ],
@@ -200,8 +198,8 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "2022-11-16 11:23:55.611091: I tensorflow/stream_executor/cuda/cuda_dnn.cc:366] Loaded cuDNN version 8401\n",
-      "2022-11-16 11:23:56.047993: I tensorflow/core/platform/default/subprocess.cc:304] Start cannot spawn child process: No such file or directory\n"
+      "2022-11-16 11:31:03.488364: I tensorflow/stream_executor/cuda/cuda_dnn.cc:366] Loaded cuDNN version 8401\n",
+      "2022-11-16 11:31:03.969312: I tensorflow/core/platform/default/subprocess.cc:304] Start cannot spawn child process: No such file or directory\n"
      ]
     },
     {
@@ -219,7 +217,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 8,
    "metadata": {},
    "outputs": [
     {
@@ -227,26 +225,58 @@
      "output_type": "stream",
      "text": [
       "at epoch 1\n",
-      "train info: logloss loss:358.075682867141\n",
-      "eval info: auc:0.729, logloss:0.5033\n",
-      "at epoch 1 , train time: 27.8 eval time: 4.1\n",
+      "train info: logloss loss:744.3602104187012\n",
+      "eval info: auc:0.6637, logloss:0.5342\n",
+      "at epoch 1 , train time: 20.7 eval time: 4.0\n",
       "at epoch 2\n",
-      "train info: logloss loss:20.122149774006434\n",
-      "eval info: auc:0.74, logloss:0.4971\n",
-      "at epoch 2 , train time: 27.5 eval time: 4.1\n",
+      "train info: logloss loss:385.66929054260254\n",
+      "eval info: auc:0.7137, logloss:0.5109\n",
+      "at epoch 2 , train time: 19.9 eval time: 3.9\n",
       "at epoch 3\n",
-      "train info: logloss loss:1.1374033954408433\n",
-      "eval info: auc:0.7419, logloss:0.4959\n",
-      "at epoch 3 , train time: 27.9 eval time: 4.1\n",
+      "train info: logloss loss:191.5083179473877\n",
+      "eval info: auc:0.7283, logloss:0.5037\n",
+      "at epoch 3 , train time: 19.7 eval time: 4.1\n",
       "at epoch 4\n",
-      "train info: logloss loss:0.5466613788453359\n",
-      "eval info: auc:0.742, logloss:0.4961\n",
-      "at epoch 4 , train time: 27.9 eval time: 4.1\n",
+      "train info: logloss loss:92.20774817466736\n",
+      "eval info: auc:0.7359, logloss:0.4991\n",
+      "at epoch 4 , train time: 20.1 eval time: 3.9\n",
       "at epoch 5\n",
-      "train info: logloss loss:0.5320256811285776\n",
-      "eval info: auc:0.742, logloss:0.4964\n",
-      "at epoch 5 , train time: 28.1 eval time: 4.1\n"
+      "train info: logloss loss:43.15945792198181\n",
+      "eval info: auc:0.74, logloss:0.4963\n",
+      "at epoch 5 , train time: 20.0 eval time: 3.9\n",
+      "at epoch 6\n",
+      "train info: logloss loss:19.656923294067383\n",
+      "eval info: auc:0.7426, logloss:0.4946\n",
+      "at epoch 6 , train time: 20.3 eval time: 3.9\n",
+      "at epoch 7\n",
+      "train info: logloss loss:8.770357608795166\n",
+      "eval info: auc:0.7441, logloss:0.4934\n",
+      "at epoch 7 , train time: 19.9 eval time: 4.0\n",
+      "at epoch 8\n",
+      "train info: logloss loss:3.9227356910705566\n",
+      "eval info: auc:0.7453, logloss:0.4925\n",
+      "at epoch 8 , train time: 19.8 eval time: 4.1\n",
+      "at epoch 9\n",
+      "train info: logloss loss:1.859877161681652\n",
+      "eval info: auc:0.7462, logloss:0.4917\n",
+      "at epoch 9 , train time: 20.2 eval time: 4.0\n",
+      "at epoch 10\n",
+      "train info: logloss loss:1.0249397866427898\n",
+      "eval info: auc:0.747, logloss:0.491\n",
+      "at epoch 10 , train time: 20.2 eval time: 4.0\n",
+      "CPU times: user 3min 57s, sys: 8.46 s, total: 4min 5s\n",
+      "Wall time: 4min\n"
      ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7f861579ebd0>"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
@@ -256,9 +286,17 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 9,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'auc': 0.7356, 'logloss': 0.5017}\n"
+     ]
+    }
+   ],
    "source": [
     "# check the predictive performance after the model is trained\n",
     "result = model.run_eval(test_file)\n",
@@ -267,16 +305,38 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 10,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "data": {
+      "application/scrapbook.scrap.json+json": {
+       "data": {
+        "auc": 0.7356,
+        "logloss": 0.5017
+       },
+       "encoder": "json",
+       "name": "result",
+       "version": 1
+      }
+     },
+     "metadata": {
+      "scrapbook": {
+       "data": true,
+       "display": false,
+       "name": "result"
+      }
+     },
+     "output_type": "display_data"
+    }
+   ],
    "source": [
     "sb.glue(\"result\", result)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 11,
    "metadata": {},
    "outputs": [],
    "source": [

--- a/examples/00_quick_start/xdeepfm_criteo.ipynb
+++ b/examples/00_quick_start/xdeepfm_criteo.ipynb
@@ -46,21 +46,19 @@
     }
    ],
    "source": [
-    "import sys\n",
     "import os\n",
+    "import sys\n",
     "import scrapbook as sb\n",
     "from tempfile import TemporaryDirectory\n",
     "import tensorflow as tf\n",
     "tf.get_logger().setLevel('ERROR') # only show error messages\n",
     "\n",
-    "from recommenders.models.deeprec.deeprec_utils import (\n",
-    "    download_deeprec_resources, prepare_hparams\n",
-    ")\n",
+    "from recommenders.models.deeprec.deeprec_utils import download_deeprec_resources, prepare_hparams\n",
     "from recommenders.models.deeprec.models.xDeepFM import XDeepFMModel\n",
     "from recommenders.models.deeprec.io.iterator import FFMTextIterator\n",
     "\n",
     "print(\"System version: {}\".format(sys.version))\n",
-    "print(\"Tensorflow version: {}\".format(tf.__version__))\n"
+    "print(\"Tensorflow version: {}\".format(tf.__version__))"
    ]
   },
   {
@@ -103,7 +101,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "100%|██████████| 10.3k/10.3k [00:01<00:00, 6.89kKB/s]\n"
+      "100%|██████████| 10.3k/10.3k [00:01<00:00, 7.08kKB/s]\n"
      ]
     }
    ],
@@ -133,7 +131,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 13,
+   "execution_count": 5,
    "metadata": {},
    "outputs": [
     {
@@ -153,20 +151,19 @@
     "                          embed_l2=0.01, \n",
     "                          layer_l2=0.01,\n",
     "                          learning_rate=0.002, \n",
-    "                          batch_size=BATCH_SIZE_CRITEO, \n",
-    "                          epochs=EPOCHS_FOR_CRITEO_RUN, \n",
+    "                          batch_size=BATCH_SIZE, \n",
+    "                          epochs=EPOCHS, \n",
     "                          cross_layer_sizes=[20, 10], \n",
     "                          init_value=0.1, \n",
     "                          layer_sizes=[20,20],\n",
     "                          use_Linear_part=True, \n",
     "                          use_CIN_part=True, \n",
-    "                          use_DNN_part=True)\n",
-    "\n"
+    "                          use_DNN_part=True)\n"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 16,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [
     {
@@ -182,7 +179,11 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "2022-11-16 11:12:42.279935: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 15397 MB memory:  -> device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0001:00:00.0, compute capability: 6.0\n"
+      "/anaconda/envs/reco_gpu/lib/python3.7/site-packages/tensorflow/python/framework/indexed_slices.py:450: UserWarning: Converting sparse IndexedSlices(IndexedSlices(indices=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Reshape_1:0\", shape=(None,), dtype=int32), values=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Reshape:0\", shape=(None, 10), dtype=float32), dense_shape=Tensor(\"gradients/XDeepFM/embedding/embedding_lookup_sparse/GatherV2_grad/Cast:0\", shape=(2,), dtype=int32))) to a dense Tensor of unknown shape. This may consume a large amount of memory.\n",
+      "  \"shape. This may consume a large amount of memory.\" % value)\n",
+      "2022-11-16 11:23:52.751265: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
+      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
+      "2022-11-16 11:23:54.342328: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 15397 MB memory:  -> device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0001:00:00.0, compute capability: 6.0\n"
      ]
     }
    ],
@@ -192,9 +193,17 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 17,
+   "execution_count": 7,
    "metadata": {},
    "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-11-16 11:23:55.611091: I tensorflow/stream_executor/cuda/cuda_dnn.cc:366] Loaded cuDNN version 8401\n",
+      "2022-11-16 11:23:56.047993: I tensorflow/core/platform/default/subprocess.cc:304] Start cannot spawn child process: No such file or directory\n"
+     ]
+    },
     {
      "name": "stdout",
      "output_type": "stream",
@@ -210,7 +219,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 18,
+   "execution_count": null,
    "metadata": {},
    "outputs": [
     {
@@ -218,56 +227,26 @@
      "output_type": "stream",
      "text": [
       "at epoch 1\n",
-      "train info: logloss loss:744.3602104187012\n",
-      "eval info: auc:0.6637, logloss:0.5342\n",
-      "at epoch 1 , train time: 20.0 eval time: 3.9\n",
+      "train info: logloss loss:358.075682867141\n",
+      "eval info: auc:0.729, logloss:0.5033\n",
+      "at epoch 1 , train time: 27.8 eval time: 4.1\n",
       "at epoch 2\n",
-      "train info: logloss loss:385.66929054260254\n",
-      "eval info: auc:0.7137, logloss:0.5109\n",
-      "at epoch 2 , train time: 19.2 eval time: 4.0\n",
+      "train info: logloss loss:20.122149774006434\n",
+      "eval info: auc:0.74, logloss:0.4971\n",
+      "at epoch 2 , train time: 27.5 eval time: 4.1\n",
       "at epoch 3\n",
-      "train info: logloss loss:191.5083179473877\n",
-      "eval info: auc:0.7283, logloss:0.5037\n",
-      "at epoch 3 , train time: 19.5 eval time: 3.9\n",
+      "train info: logloss loss:1.1374033954408433\n",
+      "eval info: auc:0.7419, logloss:0.4959\n",
+      "at epoch 3 , train time: 27.9 eval time: 4.1\n",
       "at epoch 4\n",
-      "train info: logloss loss:92.20774817466736\n",
-      "eval info: auc:0.7359, logloss:0.4991\n",
-      "at epoch 4 , train time: 19.4 eval time: 3.9\n",
+      "train info: logloss loss:0.5466613788453359\n",
+      "eval info: auc:0.742, logloss:0.4961\n",
+      "at epoch 4 , train time: 27.9 eval time: 4.1\n",
       "at epoch 5\n",
-      "train info: logloss loss:43.15945792198181\n",
-      "eval info: auc:0.74, logloss:0.4963\n",
-      "at epoch 5 , train time: 19.6 eval time: 3.9\n",
-      "at epoch 6\n",
-      "train info: logloss loss:19.656923294067383\n",
-      "eval info: auc:0.7426, logloss:0.4946\n",
-      "at epoch 6 , train time: 19.7 eval time: 3.9\n",
-      "at epoch 7\n",
-      "train info: logloss loss:8.770357578992844\n",
-      "eval info: auc:0.7441, logloss:0.4934\n",
-      "at epoch 7 , train time: 19.6 eval time: 4.0\n",
-      "at epoch 8\n",
-      "train info: logloss loss:3.9227356910705566\n",
-      "eval info: auc:0.7453, logloss:0.4925\n",
-      "at epoch 8 , train time: 19.8 eval time: 3.9\n",
-      "at epoch 9\n",
-      "train info: logloss loss:1.8598771467804909\n",
-      "eval info: auc:0.7462, logloss:0.4917\n",
-      "at epoch 9 , train time: 19.5 eval time: 4.0\n",
-      "at epoch 10\n",
-      "train info: logloss loss:1.0249397829174995\n",
-      "eval info: auc:0.747, logloss:0.491\n",
-      "at epoch 10 , train time: 19.7 eval time: 3.9\n"
+      "train info: logloss loss:0.5320256811285776\n",
+      "eval info: auc:0.742, logloss:0.4964\n",
+      "at epoch 5 , train time: 28.1 eval time: 4.1\n"
      ]
-    },
-    {
-     "data": {
-      "text/plain": [
-       "<recommenders.models.deeprec.models.xDeepFM.XDeepFMModel at 0x7fa2146f26d0>"
-      ]
-     },
-     "execution_count": 18,
-     "metadata": {},
-     "output_type": "execute_result"
     }
    ],
    "source": [
@@ -277,17 +256,9 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 19,
+   "execution_count": null,
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "{'auc': 0.7356, 'logloss': 0.5017}\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
     "# check the predictive performance after the model is trained\n",
     "result = model.run_eval(test_file)\n",
@@ -296,38 +267,16 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 18,
+   "execution_count": null,
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "application/scrapbook.scrap.json+json": {
-       "data": {
-        "auc": 0.7356,
-        "logloss": 0.5017
-       },
-       "encoder": "json",
-       "name": "res_real",
-       "version": 1
-      }
-     },
-     "metadata": {
-      "scrapbook": {
-       "data": true,
-       "display": false,
-       "name": "res_real"
-      }
-     },
-     "output_type": "display_data"
-    }
-   ],
+   "outputs": [],
    "source": [
     "sb.glue(\"result\", result)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 19,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [

--- a/tests/integration/examples/test_notebooks_gpu.py
+++ b/tests/integration/examples/test_notebooks_gpu.py
@@ -164,15 +164,11 @@ def test_fastai_integration(
 @pytest.mark.notebooks
 @pytest.mark.integration
 @pytest.mark.parametrize(
-    "syn_epochs, criteo_epochs, expected_values, seed",
+    "epochs, expected_values, seed",
     [
         (
-            15,
-            10,
-            {
-                "res_syn": {"auc": 0.9716, "logloss": 0.699},
-                "res_real": {"auc": 0.749, "logloss": 0.4926},
-            },
+            5,
+            {"auc": 0.742, "logloss": 0.4964},
             42,
         )
     ],
@@ -181,8 +177,7 @@ def test_xdeepfm_integration(
     notebooks,
     output_notebook,
     kernel_name,
-    syn_epochs,
-    criteo_epochs,
+    epochs,
     expected_values,
     seed,
 ):
@@ -192,10 +187,8 @@ def test_xdeepfm_integration(
         output_notebook,
         kernel_name=kernel_name,
         parameters=dict(
-            EPOCHS_FOR_SYNTHETIC_RUN=syn_epochs,
-            EPOCHS_FOR_CRITEO_RUN=criteo_epochs,
-            BATCH_SIZE_SYNTHETIC=1024,
-            BATCH_SIZE_CRITEO=1024,
+            EPOCHS=epochs,
+            BATCH_SIZE=1024,
             RANDOM_SEED=seed,
         ),
     )

--- a/tests/integration/examples/test_notebooks_gpu.py
+++ b/tests/integration/examples/test_notebooks_gpu.py
@@ -197,11 +197,7 @@ def test_xdeepfm_integration(
     ]
 
     for key, value in expected_values.items():
-        assert results[key]["auc"] == pytest.approx(value["auc"], rel=TOL, abs=ABS_TOL)
-        assert results[key]["logloss"] == pytest.approx(
-            value["logloss"], rel=TOL, abs=ABS_TOL
-        )
-
+        assert results[key] == pytest.approx(value, rel=TOL, abs=ABS_TOL)
 
 @pytest.mark.gpu
 @pytest.mark.notebooks

